diff --git a/scripts/vitals.py b/scripts/vitals.py
index 812a01b..5d0770a 100644
--- a/scripts/vitals.py
+++ b/scripts/vitals.py
@@ -12,7 +12,6 @@ import os
 from obspy import UTCDateTime
 import plotly.graph_objs as graph
 import plotly.offline as plotly
-from pdb import set_trace as keyboard
 
 # Get current version number
 version = setup.get_version()
@@ -249,10 +248,10 @@ def write_corrected_pressure_offset(dives_dict, processed_path):
             warn_str = ''
             if lastdive.p2t_offset_corrected > 200:
                 warn_str = '!!!'
-                print("\n\n!!! WARNING: {:s} corrected external pressure was {:d} mbar at last surfacing !!!"
+                print("\n!!! WARNING: {:s} corrected external pressure was {:d} mbar at last surfacing"
                       .format(mfloat, lastdive.p2t_offset_corrected))
-                print("!!! The corrected external pressure must stay below 300 mbar !!!")
-                print("!!! Consider adjusting {:s}.cmd using 'p2t qm!offset ...' AFTER 'buoy bypass' and BEFORE 'stage ...' !!!\n\n"
+                print("!!! The corrected external pressure must stay below 300 mbar")
+                print("!!! Consider adjusting {:s}.cmd using 'p2t qm!offset ...' AFTER 'buoy bypass' and BEFORE 'stage ...'\n"
                       .format(mfloat))
 
             f.write(lastdive_fmt_spec.format(mfloat,
