diff --git a/scripts/main.py b/scripts/main.py
index 34781d3..90f3717 100644
--- a/scripts/main.py
+++ b/scripts/main.py
@@ -4,7 +4,7 @@
 # Original author: Sebastien Bonnieux
 # Current maintainer: Dr. Joel D. Simon (JDS)
 # Contact: jdsimon@alumni.princeton.edu | joeldsimon@gmail.com
-# Last modified by JDS: 05-Oct-2020, Python 2.7.15, Darwin-18.7.0-x86_64-i386-64bit
+# Last modified by JDS: 14-Oct-2020, Python 2.7.15, Darwin-18.7.0-x86_64-i386-64bit
 
 import setup
 import os
@@ -29,8 +29,10 @@ def_processed_path = os.path.join(def_mermaid_path, "processed")
 
 # Parse (optional) command line inputs to override default paths
 parser = argparse.ArgumentParser()
-parser.add_argument('--server', default=def_server_path, help="Path to server directory")
-parser.add_argument('--processed', default=def_processed_path, help="Path to processed directory")
+parser.add_argument('--server', default=def_server_path,
+                    help="path to server directory (default: {:s})".format(def_server_path))
+parser.add_argument('--processed', default=def_processed_path,
+                    help="path to processed directory (default: {:s})".format(def_processed_path))
 args = parser.parse_args()
 server_path = os.path.abspath(args.server)
 processed_path = os.path.abspath(args.processed)
@@ -150,11 +152,6 @@ def main():
               .format(mfloat_serial))
         mdives = dives.get_dives(mfloat_path, mevents)
 
-        # Attach a completeness metric concerning the data in the .MER file to
-        # each event -- this is only possible after collecting all dives
-        # (mdives) and all events separately (mevents).
-        dives.attach_mmd_is_complete_to_dive_events(mdives)
-
         # Compute files for each dive
         for dive in mdives:
             # Create the directory
