diff --git a/scripts/setup.py b/scripts/setup.py
index 5d2a2ef..9bdf47e 100644
--- a/scripts/setup.py
+++ b/scripts/setup.py
@@ -3,8 +3,8 @@
 #
 # Author: Dr. Joel D. Simon
 # Contact: jdsimon@alumni.princeton.edu | joeldsimon@gmail.com
-# Last modified: 01-Oct-2020, Python 2.7.15, Darwin-18.7.0-x86_64-i386-64bit
+# Last modified: 05-Oct-2020, Python 2.7.15, Darwin-18.7.0-x86_64-i386-64bit
 
 def get_version():
-    version = 'v1.2.0'
+    version = 'v1.3.0'
     return version
