diff --git a/scripts/tool_invert_mer.py b/scripts/tool_invert_mer.py
index 14cad22..8e6a2de 100644
--- a/scripts/tool_invert_mer.py
+++ b/scripts/tool_invert_mer.py
@@ -10,7 +10,6 @@
 #          the entire time difference between using this tool and main.py...
 #
 # Original author: Sebastien Bonnieux
-#
 # Current maintainer: Dr. Joel D. Simon (JDS)
 # Contact: jdsimon@alumni.princeton.edu | joeldsimon@gmail.com
 # Last modified by JDS: 01-Oct-2020, Python 2.7.15, Darwin-18.7.0-x86_64-i386-64bit
@@ -25,7 +24,7 @@ from pdb import set_trace as keyboard
 # Get current version number.
 version = setup.get_version()
 
-mmd_file_name = sys.argv[1];
+mmd_file_name = sys.argv[1]
 mmd_file_path = os.path.join(os.path.split(mmd_file_name)[0], "")
 
 def invert_main():
