diff --git a/scripts/events.py b/scripts/events.py
index f955d5f..3e803e1 100644
--- a/scripts/events.py
+++ b/scripts/events.py
@@ -434,14 +434,18 @@ class Event:
 
 
 def write_traces_txt(mdives, processed_path, mfloat_path, mfloat):
-    fmt_spec = '{:>40s}    {:>15s}    {:>15s}    {:>15s}    {:>15s}    {:>15s}\n'
+    fmt_spec = '{:>40s}    {:>15s}    {:>15s}    {:>15s}    {:>15s}    {:>15s}    {:>15s}    {:>15s}\n'
     traces_file = os.path.join(processed_path, mfloat_path, mfloat+"_traces.txt")
     event_dive_tup = ((event, dive) for dive in mdives for event in dive.events if event.station_loc)
+
     with open(traces_file, "w+") as f:
-        f.write("            	         SAC_MSEED_TRACE            BIN_MER      THIS_DIVE_LOG  THIS_DIVE_ENV_MER      NEXT_DIVE_LOG  NEXT_DIVE_ENV_MER\n".format())
+        f.write("            	         SAC_MSEED_TRACE            BIN_MER      PREV_DIVE_LOG  PREV_DIVE_ENV_MER      THIS_DIVE_LOG  THIS_DIVE_ENV_MER      NEXT_DIVE_LOG  NEXT_DIVE_ENV_MER\n".format())
+
         for e, d in sorted(event_dive_tup, key=lambda x: x[0].date):
             f.write(fmt_spec.format(e.get_export_file_name(),
                                     e.mer_binary_name,
+                                    d.prev_dive_log_name,
+                                    d.prev_dive_mer_environment_name,
                                     d.log_name,
                                     d.mer_environment_name,
                                     d.next_dive_log_name,
