diff --git a/scripts/main.py b/scripts/main.py
index 3f24209..85eb3a9 100644
--- a/scripts/main.py
+++ b/scripts/main.py
@@ -169,6 +169,8 @@ def main():
         # Correlate the list of events with each dive.
         print(" ...matching those events to {:s} .LOG ('dive') files (GPS & dive metadata)..." \
               .format(mfloat_serial))
+
+        # Really: collect all the .LOG files in order (1 .LOG == 1 Dive)
         mdives = dives.get_dives(mfloat_path, mevents)
 
         # Compute files for each dive
@@ -183,15 +185,16 @@ def main():
             # Generate dive plot
             dive.generate_dive_plotly()
 
-        # Compute clock drift correction for each event, and build list of GPS locations.
+        # Compute clock drift correction for each event, and build list of GPS locations
         for dive in mdives:
             dive.correct_events_clockdrift()
 
-        # Compute location of mermaid float for each event (because the station is moving)
-        # the algorithm use gps information in the next dive to estimate surface drift
-        i = 0
+        # Interpolate for the locations that MERMAID passed out of/in to the surface and mixed
+        # layers, and where it was when it recorded any events associated with the dive
+        mdives[0].compute_station_locations(None, mdives[1], mixed_layer_depth_m)
+        i = 1
         while i < len(mdives)-1:
-            mdives[i].compute_station_locations(mdives[i+1], mixed_layer_depth_m)
+            mdives[i].compute_station_locations(mdives[i-1], mdives[i+1], mixed_layer_depth_m)
             i += 1
 
         # Generate plots, SAC, and miniSEED files
